python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -220,6 +232,43 @@ class NearestNeighbors(Base):
         self.metric_params = metric_params
         self.p = p
         self.algorithm = algorithm
+        self.algo_params = algo_params
+
+    @staticmethod
+    def _check_algo_params(algo, params):
+        if params is None:
+            return
+        if 'automated' in params and params['automated']:
+            if algo == "PQ":
+                for param in ['nlist', 'M', 'n_bits', 'usePrecomputedTables']:
+                    if not hasattr(params, param):
+                        ValueError('algo_params misconfigured : {} \
+                                    parameter unset'.format(param))
+
+    @staticmethod
+    def _build_algo_params(algo, params):
+        NearestNeighbors._check_algo_params(algo, params)
+        if params is None:
+            params = dict({'automated': True})
+
+        cdef knnIndexParam* algo_params = new knnIndexParam()
+
+        if algo == 'brute':
+            algo_params.type = Bruteforce
+        elif algo == 'PQ':
+            algo_params.type = PQ
+
+        automated = 'automated' in params and params['automated']
```

Comment:
Can you explain how this option is going to work?

Reply:
If `algo_params` is set to `None` or to a dictionary with key `'automated'` set to `True`, then the chosen strategy will be launched with a default set of parameters. These parameters are generated automatically knowing the chosen method and the specificities of the input. (see `approx_knn_ivfpq_build_index`)

Reply:
That makes sense. Thanks.